source/system/structs/HashMap.ooc

```diff
@@ -20,6 +23,12 @@ _HashEntry: class <K, V> {
 		}
 		super()
 	}
+	freeKeyValuePair: func {
+		if (K inheritsFrom(Object))
+			(this _key as Object) free()
+		if (V inheritsFrom(Object))
+			(this _value as Object) free()
+	}
```

Comment:
Thanks @sebastianbaginski - this works great (didn't even need the `*@` pointer bonanza).